src/basics/Player.js

```diff
@@ -125,8 +125,13 @@ export class Player {
 				(state.strikes === 2 ||
 					card.possible.some(p => state.play_stacks[p.suitIndex] + 1 < p.rank && p.rank <= state.max_ranks[p.suitIndex]) ||
 					Array.from(linked_orders).some(o => this.thoughts[o].focused && o !== c.order));
+			// We are waiting for a play to prove this connection
+			const waiting_proof = this.waiting_connections.some(wc =>
+				wc.connections.some((conn, ci) => ci >= wc.conn_index && conn.card.order === c.order &&
+					wc.connections.some((conn2, ci2) => ci2 >= wc.conn_index && ci > ci2 && conn2.type == 'waiting'))
+			);
 
-			return (!card.trash || card.possible.every(p => !state.isBasicTrash(p))) && !unsafe_linked &&
+			return (!card.trash || card.possible.every(p => !state.isBasicTrash(p))) && !unsafe_linked && !waiting_proof &&
 				card.possibilities.every(p => (card.chop_moved ? state.isBasicTrash(p) : false) || state.isPlayable(p)) &&	// cm cards can ignore trash ids
 				card.possibilities.some(p => state.isPlayable(p)) &&	// Exclude empty case
```

Comment:
It feels a bit odd to me that *possibilities* has g1 as the only possible card. I thought only inferred was supposed to include assumptions like this?

Reply:
`Card.possibilities` is different from `Card.possible`: the former is an alias for `Card.inferred` if there is at least one inference, otherwise it is an alias for `Card.possible`. This ensures that fully-known cards that have lost all their inferences multiple times (due to mistakes, for example) are still correctly recognized to be playable/trash.